tests/js/client/shell/shell-transaction-intermediate-commit-cluster.js

```diff
@@ -0,0 +1,491 @@
+      }
+      
+      // follower must have been dropped
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_count");
+      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+    
+      let intermediateCommitsAfter = getMetric(follower, "arangodb_intermediate_commits");
+      assertEqual(intermediateCommitsBefore + 10, intermediateCommitsAfter);
+      
+      assertInSync(leader, follower, shardId);
+    },
+    
+    // make follower execute intermediate commits (before the leader), and let the
+    // transaction fail
+    testSingleStreamingIntermediateCommitsOnFollowerWithRollback: function () {
+      db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      let follower = getEndpointById(shards[shardId][1]);
+      // disable intermediate commits on leader
+      debugSetFailAt(leader, "noIntermediateCommits");
+      // turn on intermediate commits on follower
+      debugClearFailAt(follower, "noIntermediateCommits");
+      
+      const opts = {
+        collections: {
+          write: [cn]
+        },
+        intermediateCommitCount: 1000,
+        options: { intermediateCommitCount: 1000 }
+      };
+      
+      let droppedFollowersBefore = getMetric(leader, "arangodb_dropped_followers_count");
+      let intermediateCommitsBefore = getMetric(follower, "arangodb_intermediate_commits");
+      
+      const trx = db._createTransaction(opts);
+      const tc = trx.collection(cn);
+      for (let i = 0; i < 9999; ++i) {
+        tc.insert({ _key: "test" + i });
+      }
+      trx.abort();
+      
+      // follower must have been dropped
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_count");
+      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+    
+      let intermediateCommitsAfter = getMetric(follower, "arangodb_intermediate_commits");
+      assertEqual(intermediateCommitsBefore + 9, intermediateCommitsAfter);
+      
+      assertInSync(leader, follower, shardId);
+    },
+    
+  };
+}
+
+function transactionIntermediateCommitsMultiSuite() {
+  'use strict';
+  const cn = 'UnitTestsTransaction';
+
+  return {
+
+    setUp: function () {
+      getEndpointsByType("dbserver").forEach((ep) => debugClearFailAt(ep));
+      db._drop(cn + "1");
+      db._drop(cn + "2");
+    },
+
+    tearDown: function () {
+      getEndpointsByType("dbserver").forEach((ep) => debugClearFailAt(ep));
+      db._drop(cn + "1");
+      db._drop(cn + "2");
+    },
+    
+    // make follower execute intermediate commits (before the leader), but let the
+    // transaction succeed
+    testMultiAqlIntermediateCommitsOnFollower: function () {
+      db._create(cn + "1", { numberOfShards: 1, replicationFactor: 2 });
+      let shards1 = db._collection(cn + "1").shards(true);
```

Comment:
I think this creation of collections stunt should be a helper function.

Also I think you can drastically increase the chances by setting the `avoidServers` option on the second collection.
This way you can avoid placing any shard on the leader of collection `cn1`. effectively granting a 50/50 chance.

Reply:
`avoidServers` will also avoid using the server for follower shards